### DISCUSSION TITLE
Nested Table that selects across multiple nested tables

### DISCUSSION THREAD

**Discussion by WeiTangLau** [TanStack/table#2953] [Q&A]
I want to implement a nested table that supports selection from the nested table.
e.g. top layer is select product, 2nd layer is select color.
Given this example, I can select say iPhone. Following which will expand a sub table with all the different colors. I can select say black and white. After which, I select Galaxy S20 at another top layer. I then process to select gray.

This will generate me an array of 3 selected products (objects).

Is this possible? Please let me know If I did not make myself clear

**Comment by automatedmachine** [comment]
I *think* you are looking for row grouping. See this example, where you can click the "fist" on a header to group by that column. You can then expand/collapse the rows for the group.
[https://react-table.tanstack.com/docs/examples/grouping](https://react-table.tanstack.com/docs/examples/grouping)

If you are looking for something else you might want to include a basic illustration.

**Reply by WeiTangLau** [reply]
Thanks for your response @automatedmachine . Happy new year to you!
I see what you mean there. Yes the intended result can be achieved by the row grouping. However, in my use case, to better improve the user experience, I would need to have 2-level table.

Top Layer -> Filter by Phone
Second Layer -> Shows all the available colors for that filtered phone.

Using this [link](https://codesandbox.io/s/github/tannerlinsley/react-table/tree/master/examples/sub-components) as example, after selecting the "phone", I click on the 👉, it should show another table as sub-component. Furthermore, I will still need to select items across multiple sub-table. 

Thank you in advance!

**Comment by aravind-korada** [comment]
Hey @WeiTangLau 

I am also trying to build  something similar to what you were trying?

Did you find any solution for that?

Thanks in advance!

**Reply by AjayShivanagol** [reply]
any updates ?

**Reply by jakehubit** [reply]
Helllo @WeiTangLau @AjayShivanagol . I am also trying to achive this and no success so far, please let us know how you managed to do it?
